src/org/labkey/test/components/react/dialogs/BulkCreateSamplesDialog.java

```diff
@@ -0,0 +1,160 @@
+        return this;
+    }
+
+    public String getQuantity()
+    {
+        return getWrapper().getFormElement(elementCache().quantity);
+    }
+
+    public BulkCreateSamplesDialog setDescription(String description)
+    {
+        getWrapper().setFormElement(elementCache().description, description);
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return getWrapper().getFormElement(elementCache().description);
+    }
+
+    public BulkCreateSamplesDialog setTextField(String fieldCaption, String value)
+    {
+        String forField = Locator.xpath("//div[@class='modal-body']//label[./span[contains(text(), '" + fieldCaption + "')]]").findElement(getComponentElement()).getAttribute("for");
+        getWrapper().setFormElement(Locator.tagWithId("input", forField), value);
+        return this;
+    }
+
+    public String getTextField(String fieldCaption)
+    {
+        String forField = Locator.xpath("//div[@class='modal-body']//label[./span[contains(text(), '" + fieldCaption + "')]]").findElement(getComponentElement()).getAttribute("for");
+        return getWrapper().getFormElement(Locator.tagWithId("input", forField));
+    }
+
+    private boolean waitForReactSelectReady(FilteringReactSelect select)
+    {
+        getWrapper().waitFor(()->!select.getSelections().contains("Loading..."), "The reactSelect field did not populate in time.", 1_000);
+        return true;
+    }
+
+    public BulkCreateSamplesDialog setSelectionField(String fieldCaption, List<String> selectValues)
+    {
+        FilteringReactSelect reactSelect = FilteringReactSelect.finder(getDriver()).followingLabelWithSpan(fieldCaption).find();
+        reactSelect.waitForLoaded(WAIT_FOR_JAVASCRIPT);
+        selectValues.forEach(s -> {reactSelect.typeAheadSelect(s);});
+        return this;
+    }
+
+    public List<String> getSelectionField(String fieldCaption)
+    {
+        FilteringReactSelect reactSelect = FilteringReactSelect.finder(getDriver()).followingLabelWithSpan(fieldCaption).find();
+        reactSelect.waitForLoaded(WAIT_FOR_JAVASCRIPT);
+        return reactSelect.getSelections();
+    }
+
+    public BulkCreateSamplesDialog setFieldWithId(String id, String value)
+    {
+        getWrapper().setFormElement(Locator.tagWithId("input", id), value);
+        return this;
+    }
+
+    public String getFieldWithId(String id)
+    {
+        return getWrapper().getFormElement(Locator.tagWithId("input", id));
+    }
+
+    public void clickAddRows()
+    {
+        elementCache().addRowsButton.click();
+
+        try
+        {
+            if (BootstrapLocators.errorBanner.findElement(getComponentElement()).isDisplayed())
+                throw new IllegalStateException("Error message on dialog: '" + BootstrapLocators.errorBanner.findElement(getComponentElement()).getText() + "'.");
+        }
+        catch(StaleElementReferenceException stale)
+        {
+            // Do nothing if stale.
+        }
+    }
```

Comment:
Wait for dialog to disappear. Maybe wait for the insert page to update somehow as well.
```suggestion
        waitForClose();
        /* Wait for insert panel to update */
    }
```